include/RAJA/policy/sycl/policy.hpp

```diff
@@ -0,0 +1,234 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   Header file containing RAJA SYCL policy definitions.
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef policy_sycl_HPP
+#define policy_sycl_HPP
+
+#include "RAJA/policy/PolicyBase.hpp"
+
+#include <cstddef>
+
+namespace RAJA
+{
+
+struct uint3 {
+  unsigned long x, y, z;
+};
+
+using sycl_dim_t = cl::sycl::range<1>;
+
+using sycl_dim_3_t = uint3;
+
+namespace detail
+{
+template <bool Async>
+struct get_launch {
+  static constexpr RAJA::Launch value = RAJA::Launch::async;
+};
+
+template <>
+struct get_launch<false> {
+  static constexpr RAJA::Launch value = RAJA::Launch::sync;
+};
+}  // end namespace detail
+
+namespace policy
+{
+namespace sycl
+{
+
+//
+//////////////////////////////////////////////////////////////////////
+//
+// Execution policies
+//
+//////////////////////////////////////////////////////////////////////
+
+template <size_t BLOCK_SIZE, bool Async = true>
+struct sycl_exec : public RAJA::make_policy_pattern_launch_platform_t<
+                       RAJA::Policy::sycl,
+                       RAJA::Pattern::forall,
+                       detail::get_launch<Async>::value,
+                       RAJA::Platform::sycl> {
+};
+
+template <size_t BLOCK_SIZE, bool Async = false>
+struct sycl_exec_nontrivial : public RAJA::make_policy_pattern_launch_platform_t<
+                       RAJA::Policy::sycl,
+                       RAJA::Pattern::forall,
+                       detail::get_launch<Async>::value,
+                       RAJA::Platform::sycl> {
+};
+
+
+struct sycl_reduce
+    : make_policy_pattern_t<RAJA::Policy::sycl, RAJA::Pattern::reduce> {
+};
+
+}  // namespace sycl
+}  // namespace policy
+
+using policy::sycl::sycl_exec;
+using policy::sycl::sycl_exec_nontrivial;
```

Comment:
Is it possible to use the compiler to make the trivial/non-trivial differentiation automatically?

Reply:
That would be most excellent, if possible.

Reply:
Yes, I had previously tried to wrap to two different types of launches in an `if constexpr` using `std::is_trivially_copyable`.  While it works on a small reproducer, when used in RAJA is it triggering a bug (tracking it).  I haven't tried to implement using `std::enable_if`, I will try it.

Reply:
Sounds good. If it works, it will be easier on users.  Please keep us posted.